### PR TITLE
ASM-4631 prefer remote datastores with most free space

### PR DESCRIPTION
**Pull request by gavin-scott**
Sort the datastores so that we prefer the remote datastore with most
free space and fall back to local datastore with most free space.

### CODE REVIEW

**Comment by gavin-scott** [comment]
This is basically the same as #67 but prefers datastore with most free space. Untested yet.

**Comment by gavin-scott** [comment]
Made a couple fixes and is working now.

File: lib/puppet/provider/vc_vm/default.rb [hunk 1]
```diff
@@ -405,36 +405,38 @@ def get_cluster_datastore
...
-          break
-        end
-      end
-      raise("No datastore found with sufficient free space") if datastore_selected.nil?
-      return "[#{datastore_selected}]"
+      datastore_selected = datastore_info.find { |d| d['free'] >= requested_size }
+      raise("No datastore found with sufficient free space") unless datastore_selected
+      Puppet.debug("Selected datastore: #{datastore_selected['name']}")
+      # Why are we putting [] around the name in this case??
+      "[#{datastore_selected['name']}]"
```
Comment (gavin-scott) [inline lib/puppet/provider/vc_vm/default.rb]: @sushilrai do you know why we are putting brackets [] around the name? It doesn't look like we are doing that in the other path where the user has specified a specific datastore name, but that was the existing code so I left it as is.

Comment (sushilrai) [inline lib/puppet/provider/vc_vm/default.rb]: RBVMOMI anticipate the Datastore path in this format, that's why I had to adjust the format.

**Comment by gavin-scott** [comment]
DO NOT MERGE. Not sure the sorting is working right, on second VM deployment it didn't pick the remote datastore with the most free space. Looking into that...

**Comment by gavin-scott** [comment]
Finished testing, this should be good to go. It correctly picks the remote datastore with the most free space and will fall back to the local datastore with the most free space.

@sushilrai can you review?

**Comment by sushilrai** [comment]
+1 Ready to be merged. If we are merging it in 8.1.1

**Pull request merged**